napari_aicssegmentation/controller/workflow_select_controller.py

```diff
@@ -75,7 +75,10 @@ def select_workflow(self, workflow_name: str):
             + workflow_name,
         )
 
-        self.model.active_workflow = self._workflow_engine.get_executable_workflow(workflow_name, layer0.data)
+        # TODO: Make sure the channel being selected is the correct dimension (when we start using AICSIMAGEIO)
+        self.model.active_workflow = self._workflow_engine.get_executable_workflow(
```

Comment:
I introduced the `LayerReader` class to make refactoring easier in the future. Let's add a `get_channel(index)` method to that class and use that instead so that all the `aicsimageio` refactoring can be done within that class later.

And while you're at it, you could consider adding the same tiny bit of logic right here https://github.com/AllenCell/napari-aicssegmentation/blob/main/napari_aicssegmentation/core/layer_reader.py#L26 to attempt guessing the channel. As silly as it is, it should actually cover a lot of cases since it's very likely that the channel would be the either the third of fourth dimension.

Reply:
A little confused on this, can you clarify?

Reply:
I'm saying we should use the `LayerReader` class (for now) to separate the logic involved in selecting a single channel out of a Napari layer, so that we can easily refactor it later when we can use `aicsimageio`. 
So instead of hard-coding this here with `layer0.data[self.model.selected_channel.index, :, :, :]`, throw this into a new method in `LayerReader`. And that new method can have some extra logic to try and figure out which dimension is the channel one, if we think it's valuable (it certainly was valuable for getting the list of channels).